### PR TITLE
Run `pre-commit` on all files

### PR DESCRIPTION
**Pull request by jparismorgan**
### What
* Follow-up to #200, where we run `pre-commit` on code in the repo.
* Makes configuration changes not modify `.ipynb` files with `ruff` or `clang-format`. This is because a) unit tests were [failing](https://github.com/TileDB-Inc/TileDB-Vector-Search/actions/runs/7556304581/job/20573027217?pr=202) with the auto fixes and b) because you don't read the source code anyways, it seems less important to get working. But happy to remove this and investigate more.

### Testing
After this change everything passes:
```
(TileDB-Vector-Search-2) ~/repo/TileDB-Vector-Search-2 pre-commit run --all-files            1 ↵  ✹main 
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
prettier.................................................................Passed
ruff.....................................................................Passed
black....................................................................Passed
clang-format.............................................................Passed
```

### CODE REVIEW

**Comment by jparismorgan** [comment]
Not yet ready for review, I missed a failing test.

File: README.md [hunk 1]
```diff
@@ -1,23 +1,21 @@
 <a href="https://tiledb.com"><img src="https://github.com/TileDB-Inc/TileDB/raw/dev/doc/source/_static/tiledb-logo_color_no_margin_@4x.png" alt="TileDB logo" width="400"></a>
 
-
-
 # TileDB Vector Search
 
-*TileDB-Vector-Search* is a C++ library and Python API for vector search built on top of the [TileDB Storage Engine](https://github.com/TileDB-Inc/TileDB).
+_TileDB-Vector-Search_ is a C++ library and Python API for vector search built on top of the [TileDB Storage Engine](https://github.com/TileDB-Inc/TileDB).
```
Comment (NikolaosPapailiou) [inline README.md:5]: Is this intended?

**Review by NikolaosPapailiou — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jparismorgan) [reply README.md:5]: Not particularly, but it appears they both result in italics when viewed as a markdown preview, and it seems to be something you can't change about prettier: https://stackoverflow.com/questions/64052818/prettier-keeps-replacing-asterisk-characters-with-underscores

**Comment by jparismorgan** [comment]
@NikolaosPapailiou re-requesting review because I had to pull from `main` and change a few more things. Going to merge now because they seem safe, but if you want to give it another review I can address concerns in a follow-up.

**Review by NikolaosPapailiou — APPROVED** [review]

**Pull request merged**